cluster-autoscaler/cloudprovider/azure/azure_scale_set.go

```diff
@@ -200,15 +202,22 @@ func (scaleSet *ScaleSet) GetScaleSetVms() ([]compute.VirtualMachineScaleSetVM,
 		return nil, err
 	}
 
+	instanceIDs := make([]string, 0)
 	for _, vm := range result {
 		instanceIDs = append(instanceIDs, *vm.InstanceID)
 	}
 
-	allVMs := make([]compute.VirtualMachineScaleSetVM, 0)
+	allVMs := make([]string, 0)
 	for _, instanceID := range instanceIDs {
+		// Get from cache first.
+		if v, ok := scaleSet.virtualMachines[instanceID]; ok {
+			allVMs = append(allVMs, v)
```

Comment:
Could we please add the logic here to ensure that the mapping is deleted when the instance is no longer present in the list ?

Reply:
Perhaps we can have the map and utility functions move there and only implement it for vmss for now and take up the rest later onwards ?

Reply:
we don't need to remove them, because in next scale up, if the instanceID is same, then resourceID would also be same.